test/common/http/conn_manager_impl_test.cc

```diff
@@ -5794,6 +5795,63 @@ TEST_F(HttpConnectionManagerImplTest, DisableKeepAliveWhenOverloaded) {
   EXPECT_EQ(1U, stats_.named_.downstream_cx_overload_disable_keepalive_.value());
 }
 
+class DrainH2HttpConnectionManagerImplTest : public HttpConnectionManagerImplTest,
+                                             public testing::WithParamInterface<bool> {
+public:
+  DrainH2HttpConnectionManagerImplTest() {
+    Runtime::LoaderSingleton::getExisting()->mergeValues(
+        {{"envoy.reloadable_features.connection_manager_drain_http2", "true"}});
+  }
+
+private:
+  TestScopedRuntime runtime_;
+};
+
+TEST_P(DrainH2HttpConnectionManagerImplTest, DisableHttp2KeepAliveWhenOverloaded) {
```

Comment:
Can you add a '//' comment above the test with a summary of what it tests? Also, I think we should have a test that covers when there are multiple in flight streams. Even better if there were an integration test somehow.